x-pack/test/cloud_security_posture_functional/cloud_tests/findings_sanity.ts

```diff
@@ -0,0 +1,169 @@
+        },
+        {
+          query:
+            'cloud.provider : "gcp" and rule.benchmark.rule_number : "3.1" and result.evaluation : "failed"',
+          provider: 'gcp',
+          expectedRowsCount: 1,
+          expectedGroupCount: '1 cloud account',
+          expectedUnitCount: '1 finding',
+        },
+        {
+          query:
+            'cloud.provider : "azure" and rule.benchmark.rule_number : "7.1" and result.evaluation : "failed"',
+          provider: 'azure',
+          expectedRowsCount: 1,
+          expectedGroupCount: '1 cloud account',
+          expectedUnitCount: '1 finding',
+        },
+        {
+          query:
+            'rule.benchmark.id : "cis_k8s" and rule.benchmark.rule_number : "4.2.4" and result.evaluation : "failed"',
+          provider: 'k8s',
+          expectedRowsCount: 2,
+          expectedGroupCount: '0 cloud accounts',
+          expectedUnitCount: '2 findings',
+        },
+        {
+          query: 'rule.benchmark.id : "cis_eks" and rule.benchmark.rule_number : "3.1.1"',
+          provider: 'eks',
+          expectedRowsCount: 2,
+          expectedGroupCount: '0 cloud accounts',
+          expectedUnitCount: '2 findings',
+        },
+      ];
+
+      testCases.forEach(
+        ({ query, provider, expectedRowsCount, expectedGroupCount, expectedUnitCount }) => {
+          it(`Querying ${provider} provider data`, async () => {
+            // Execute the query
+            await queryBar.setQuery(query);
+            await queryBar.submitQuery();
+            // Get the number of rows in the data table
+            const rowsCount = await findings
+              .createDataTableObject('latest_findings_table')
+              .getRowsCount();
+
+            // Check that the number of rows matches the expected count
+            expect(rowsCount).to.be(expectedRowsCount);
+            const groupSelector = await findings.groupSelector();
+            await groupSelector.openDropDown();
+            await groupSelector.setValue('Cloud account');
+            const grouping = await findings.findingsGrouping();
+            // Check that the group count and unit count matches the expected values
+            const groupCount = await grouping.getGroupCount();
+            expect(groupCount).to.be(expectedGroupCount);
+
+            const unitCount = await grouping.getUnitCount();
+            expect(unitCount).to.be(expectedUnitCount);
+          });
+        }
+      );
+    });
+
+    describe('Findings - Sorting data', () => {
+      afterEach(async () => {
+        const paginationBtn = await testSubjects.find('tablePaginationPopoverButton');
+        await paginationBtn.click();
+        const pageSizeOption = await testSubjects.find('tablePagination-50-rows');
+        await pageSizeOption.click();
+      });
+
+      type SortDirection = 'asc' | 'desc';
+      type TestCase = [string, string, string, SortDirection, string];
+      const paginationAndsortingTestCases: TestCase[] = [
```

Comment:
I think the approach from `Findings - Querying data` with an array of objects works better. An array of arrays is hard to read and understand what is what without locking at the iteration code